### PR TITLE
(feat) added ashbrook

### PR DESCRIPTION
**Pull request by starryburst**
new game mode which is like one-night which is a twist on traditional social deduction games like mafia

- roles currently work (i haven't tested every roles reaction to insanity though)
- a few things i want to work on before making it playable (demon/minion reveal, voting system, setup creation, a few more roles?)

### CODE REVIEW

**Comment by pikulet** [comment]
hmm, how is this different from the one night already on the site

**Comment by starryburst** [comment]
> hmm, how is this different from the one night already on the site

it is not one night

**Comment by pikulet** [comment]
hey I'm not sure about the rules for this, is there a rulebook?

**Comment by starryburst** [comment]
> 

you have 0 outcasts in your setup but you put 1 outcast in the setup

**Comment by starryburst** [comment]
> image
> 
> the game has started, the countdown has finished. but we're stuck in the lobby, we cannot play `1x doctor 5x scholar 1x deadly nightshade 1x rampager`
> 
> i've created 3 different setups. i could not play any of them, teh game is dead (it's not starting after the pregame countdown)

you have 0 outcasts in your setup but you put 1 outcast in the setup

**Comment by starryburst** [comment]
> image
> 
> [1] i think i am suppoused to be able to click the ??? Question marks, in order to see/learn the roles in this setup...?

i just tried this on closed roles on the live site and it does not work

**Comment by starryburst** [comment]
> why am i able to select roles (eg: 2x martyr, 1x possessor) if they will be 3x1x1x1 (question marks / open roles, i think?) anyway?

Screen Shot 2023-08-02 at 8 28 49 PM

in regular mafia you can add roles in alignments with 0 people

**Comment by starryburst** [comment]
so i just added the proper checks for closed role setups (the same as mafia does) so that you will be incapable of creating that setup in the first place so that error would not happen

**Comment by starryburst** [comment]
I'm just putting this here that even though the PR is in draft status it's not currently being worked on as it's waiting to be looked at by someone else (in case people aren't looking at it assuming it's still being worked on and isn't ready)

**Comment by DrSharky** [comment]
I tested it real quick, just a couple games.
I noticed the "you have used your final vote" message shows up multiple times in the system messages.
Typo in the plaguebearer description: should probably say "see" instead of "seem"
Semantic consistency issue in Starseeker description: should probably say "role" instead of "character"
Grammar typo in Starseeker description: Should probably say "player's" instead of "players".

Just a few things I noticed so far. I will test more.

**Comment by starryburst** [comment]
> I tested it real quick, just a couple games. I noticed the "you have used your final vote" message shows up multiple times in the system messages. Typo in the plaguebearer description: should probably say "see" instead of "seem" Semantic consistency issue in Starseeker description: should probably say "role" instead of "character" Grammar typo in Starseeker description: Should probably say "player's" instead of "players".
> 
> Also, when creating a setup, should the role count for villager, outcast, follower, and leader be visible? I noticed I can see that even if closed role/unique role option is not selected.
> 
> Just a few things I noticed so far. I will test more, making sure I understand the game and that it runs well. Then I'll review the code itself.

for final vote can you show me what happened? i'm not sure how to replicate it

for the role count thing the game is a closed role gamemode (there is no closed role toggle, which is why you see the role counts by default)

File: Games/types/Ashbrook/roles/cards/BecomeShadowWithPlayer.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const Random = require("../../../../../lib/Random");
+
+module.exports = class BecomeShadowWithPlayer extends Card {
+  constructor(role) {
+    super(role);
+
+    this.listeners = {
+      start: function () {
+        let goodPlayers = this.game.players.filter(
+          (p) => p.role.alignment != "Follower"
```
[the remark this answers is not in the document]
Comment (pikulet) [reply Games/types/Ashbrook/roles/cards/BecomeShadowWithPlayer.js]: player

**Pull request closed without merging**